### PR TITLE
Write docs for the issue platform

### PR DESCRIPTION
**Pull request by wedamija**

### CODE REVIEW

File: src/docs/issue-platform.mdx [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+- `noise_config`: A `NoiseConfig` that allows you to set policy on how many times a fingerprint must occur in a period of time before an issue is created
+
+Here's an example issue type:
+```python
+@dataclass(frozen=True)
+class ProfileFileIOGroupType(GroupType):
+    type_id = 2001
+    slug = "profile_file_io_main_thread"
+    description = "File I/O on Main Thread"
+    category = GroupCategory.PROFILE.value
```
Comment (phacops) [inline src/docs/issue-platform.mdx:35]: I think you should add an example of the `noise_config` in there.

Should you also document the `release` field for single tenant?

File: src/docs/issue-platform.mdx [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+If operating within the Sentry codebase, you can call <Link to="https://github.com/getsentry/sentry/blob/master/src/sentry/issues/producer.py#L28-L30">sentry.issues.produce_occurrence_to_kafka</Link> and pass in the <Link to="https://github.com/getsentry/sentry/blob/d262bbd9d8d3a2747d5251bee35c016c13ca050a/src/sentry/issues/issue_occurrence.py#L58">IssueOccurrence</Link>, and optionally the `Event` data. The fields in these are described in the schemas below.
+
+If operating from an external service then pass a json payload that matches the <Link to="#issue-occurrence-schema">Issue Occurrence Schema</Link> to the `ingest-occurrences` topic on the <Link to="https://github.com/getsentry/getsentry/blob/97039335a29517a7e48149148cdd7a30a60971e7/getsentry/conf/settings/prod.py#L567-L571">issue occurrences Kafka cluster</Link>
+
+### Issue Occurrence Schema:
+Required:
+- `id`: str. A user generated uuid in string format representing a unique id for this occurrence.
+Generally it's a good idea to generate this separately to your event id.
+- `project_id`: int. Id of the project this occurred in
+- `fingerprint`: Sequence[str].  Unique identifier for the root cause of this problem.
```
Comment (phacops) [inline src/docs/issue-platform.mdx:48]: I think you should expand into a different section how to make a fingerprint and give examples.

File: src/docs/issue-platform.mdx [hunk 3]
```diff
@@ -0,0 +1,105 @@
...
+
+### Issue Evidence Schema
+- `name`: str.
+- `value`: str
+- `important`: bool. Whether this entry contains important data to display to the user.
+One entry should be marked as important, and it will be displayed in space constrained
+integrations.
+
+### Event Schema
+The event schema should match the <Link to="/event-payloads/">Sentry event schema</Link>. Any fields/interfaces defined here can be passed along with your event. It's best to fill in as many of these as makes sense for your issue type.
```
Comment (phacops) [inline src/docs/issue-platform.mdx:80]: Link should be `/sdk/event-payloads`.

File: src/docs/issue-platform.mdx [hunk 4]
```diff
@@ -0,0 +1,105 @@
...
+
+One of:
+- `event_id`: str. The event id of an existing event, typically this would be an error or transaction id.
+- `event`: <Link to="#event-schema">Event</Link>
+
+Optional:
+- `resource_id`: str | None. An identifier used to link to an external system to load data.
+For example, this could be an id that we use to build a link to a specific product page.
+
+### Issue Evidence Schema
```
Comment (phacops) [inline src/docs/issue-platform.mdx:72]: I'd be useful to link to the JSON schema validation you have. I always found them more useful to understand a schema than a list like this. Of course, we still need the description of the fields here.

File: src/docs/issue-platform.mdx (hunk 2 above)
Comment (wedamija) [reply src/docs/issue-platform.mdx:48]: Yeah agreed, I think we will have a whole separate doc about writing detectors

File: src/docs/issue-platform.mdx (hunk 1 above)
Comment (wedamija) [reply src/docs/issue-platform.mdx:35]: Covered below in the release section, but it probably makes sense to link there from here.

**Pull request merged**